src/main/java/core/basesyntax/IsoscelesTrapezoid.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure {
+    private double topBase;
+    private double bottomBase;
+    private double leg;
+    private double baseAngle;
+
+    public IsoscelesTrapezoid(String color, double topBase, double bottomBase,
+                              double leg, double baseAngle) {
+        super(color);
+        this.topBase = topBase;
+        this.bottomBase = bottomBase;
+        this.leg = leg;
+        this.baseAngle = baseAngle;
+    }
+
+    @Override
+    public double calculateArea() {
+        return (topBase + bottomBase) / 2 * leg * Math.cos(90.0 - baseAngle);
```

Comment:
The formula for the area of an isosceles trapezoid is incorrect. The correct formula includes the height of the trapezoid, which can be calculated using the leg and the base angle. Please revise the formula to correctly calculate the area.